src/utils.jl

```diff
@@ -214,3 +214,40 @@ macro matchable(expr)
         Base.length(x::$name) = $(length(fields) + 1)
     end |> esc
 end
+
+# Check for linear coefficients
+coefficients(expr,sym::SymbolicUtils.Sym) coefficients(expr,[sym])
+function coefficients(expr,syms::AbstractArray{SymbolicUtils.Sym})
+    res = Dict{SymbolicUtils.Sym,Bool}
+    for s in syms
+        cur = Dict(s,has_nonlinear(expr,s,false))
+        merge(res,Dict)
```

Comment:
This should just be `res[s] = has_nonlinear(expr,s,false)` instead of creating a new dict in every iteration